### DISCUSSION TITLE
Beyond Real-Debrid

### DISCUSSION THREAD

**Discussion by morpheasgr** [LivingWithHippos/unchained-android#209] [General]
Been a longtime RD user but I think their passion towards evolving their service is kinda faded.

Things that kinda disappoint me:
1) No cloud "drive" - no way to access the user's cloud storage through protocols such as WebDAV, SFTP or other ways.

2) Tool/application/SaaS development is left to third-party developers like you, and afaik they don't even list your application on their site. Where is the love? IMHO they should have hired you, even if only as a freelance/part time associate. That aside, in the year 2022 the service still does not have a responsive/mobile site, the official mobile app is basically non-existent and the official browser extension does not even handle torrent/magnet associations. And for Windows users, the only advertised tool is jDownloader. Afaik jDownloader is only useful if you want to download files locally, and I don't think it supports magnets/torrents - neither regularily or utilizing debrids/seedboxes

3) Once upon a time, before magnet caching became a thing, Real-Debrid focused and was great at OCH/filehost caching/proxying. From my current experience, the number of supported hosts and stability/availability of the hosts in the service has decreased through the years.

4) Speeds/performance are a mystery to me. Mostly they are decent. But experience varies. On a 100Mbps line, with speedtests on the device in question maxing out at the time of occurence, streams that would be able to playback in real-time even with a 16Mbps line (I did the calculation and this is Megabits) end up buffering to an annoying level. Maybe RD is not the one to be blamed for this though - maybe it is being throttled by ISPs.

All that aside, most of the time I really enjoyed using their services, but I may be testing alternatives very soon.

One that comes in question is AllDebrid.
It offers webdav which is cool. The api seems okay as well.

Yet I am too pampered by your app. It's just too good.

I need your opinion on something: do you think it would be a good idea to fork your application and try to adapt it to AllDebrid's api or am I better off starting from scratch? There's quite a few libraries on Github to utilize AllDebrid's api.

**Comment by LivingWithHippos** [comment]
Real Debrid still fits my need and this app was created to learn Android so I'm not going to make a new one anytime soon. 

> Tool/application/SaaS development is left to third-party developers like you

Seeing how they let their app get removed from the play store (I guess they didn't update it when some google policy changed) I think they just stopped caring about anything that isn't the main web service. Still, I think all the kodi addons demonstrate that the service with the apis is still good.

> One that comes in question is AllDebrid

I know of this and Premiumize


> do you think it would be a good idea to fork your application and try to adapt it to AllDebrid's api or am I better off starting from scratch?

How much do you know about kotlin/android? You could start from this repo but between learning how to code and understanding how Unchained is made you risk losing some time... I'd "cannibalize" Unchained but I know how it works, maybe as you I'd start from scratch. 

It's in the plans to make a kotlin library but definitely not in the short term, that way it may be easier to re use it. 

ANYWAY if you ever decide to build something start from a kotlin (or java) library to interact with alldebrid, that way more people would contribute to it because it could be used for different projects like telegram bots, desktop application et cetera. You could then just plug it with an android project.

**Comment by LivingWithHippos** [comment]
> There's quite a few libraries on Github to utilize AllDebrid's api.

Not for java/kotlin sadly. You could probably save quite a lot of time using the code generator with the alldebrid openapi specification

https://pierre-emmanuelj.github.io/open-alldebrid/

https://github.com/OpenAPITools/openapi-generator
